File: 总题库/128.LongestConsecutiveSequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> seen;
        int maxLength = 0;
        for (int i : nums) {
            seen.insert(i);
        }
        for (auto iter = seen.begin(); iter != seen.end(); iter++) {
            if (seen.count(*iter - 1)) continue;
            int num = *iter;
            int tempLength = 0;
            while (seen.count(num)) {
                ++tempLength;
                ++num;
            }
            maxLength = max(maxLength, tempLength);
        }
        return maxLength;
    }
};

File: 总题库/128.LongestConsecutiveSequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        seen = set()
        maxLen = 0
        for num in nums:
            seen.add(num)
        
        for num in seen:
            if num - 1 in seen:
                continue

            tempLen = 0
            while num in seen:
                tempLen += 1
                num += 1
            maxLen = max(maxLen, tempLen)
        return maxLen